org.scala-ide.sdt.debug/src/org/scalaide/debug/internal/async/StepMessageOut.scala

```diff
@@ -50,38 +59,57 @@ case class StepMessageOut(debugTarget: ScalaDebugTarget, thread: ScalaThread) ex
     thread.resumeFromScala(DebugEvent.CLIENT_REQUEST)
   }
 
+  private[async] def findThisType(stepEvent: StepEvent): String = {
+    val loc = stepEvent.location()
+    val declType = loc.declaringType().name()
+    Try {
+      val thisObject = stepEvent.thread().frame(0).thisObject()
+      if (thisObject ne null)
+        thisObject.referenceType().name()
+      else
+        declType
+    }.recover {
+      case e: Throwable =>
+        declType
+    }.get
+  }
+
   object subordinate extends BaseDebuggerActor {
     import scala.collection.JavaConverters._
-    private def isReceiveHandler(loc: com.sun.jdi.Location): Boolean =
-      loc.method().name().contains("applyOrElse")
+    private def isReceiveHandler(stepEvent: StepEvent): Boolean = {
+      val thisType = findThisType(stepEvent)
+      stepEvent.location.method.name.contains("applyOrElse") &&
```

Comment:
well, I think that there was something like trial and error in the past. So looks like 'apply' was considered too... Now looks like torpidity. I try it and replace